scripts/build_scripts/build_script_all_ssl.sh

```diff
@@ -0,0 +1,318 @@
+#!/bin/bash
+
+# This script assumes that you have checked out the release / version
+# that you want to build.
+# This script must be run from the <project root>/scripts/build_scripts directory
+
+cd ../../
+BUILD_ROOT=`pwd`
+echo The build root is $BUILD_ROOT
+
+# Detect the platform (similar to $OSTYPE)
+# see http://stackoverflow.com/questions/394230/detect-the-os-from-a-bash-script
+# We have only three build platform variants: windows, linux, osx.
+lowercase(){
+    echo "$1" | sed "y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/"
+}
+
+OS=`lowercase \`uname\``
+# OS="`uname`"
+UN2_BUILD_OS="platform_undetected"
+case $OS in
+  linux | freebsd)
+    OS='Linux'
+    UN2_BUILD_OS="linux"
+    ;;
+  mingw* | msys* | cygwin | windowsnt)
+    OS='Windows'
+    UN2_BUILD_OS="windows"
+    ;;
+   dar*)
+    OS='Mac'
+    UN2_BUILD_OS="osx"
+    ;;
+  *) ;;
+esac
+echo Build platform is $UN2_BUILD_OS
+echo Build OS is $OS from `lowercase \`uname\``
+
+
+# reset the directory to get rid of previous build artifacts
+# Tradeoff of using git clean here, is that it also removes useful things
+# like ui/node_modules (building the UI) and modules/.
+# It also forces git to remove the tool repos which makes the build longer.
+# git clean -x -d -f -f
+rm -rf $BUILD_ROOT/bundle/
+rm -rf $BUILD_ROOT/build/
+rm -rf $BUILD_ROOT/dist/
+
+find . -type f -name '*.pyc' -exec rm -f '{}' \;
+
+if [ ! -d $BUILD_ROOT/bundle ]
+then
+  mkdir $BUILD_ROOT/bundle/
+fi
+if [ ! -d $BUILD_ROOT/bundle/static ]
+then
+  mkdir $BUILD_ROOT/bundle/static
+fi
+if [ ! -d $BUILD_ROOT/static/ui ]
+then
+  echo Making directory $BUILD_ROOT/static/ui
+  mkdir -p $BUILD_ROOT/static/ui
+else
+  echo Directory $BUILD_ROOT/static/ui exists
+fi
+
+# update the virtual environment
+pip install -r $BUILD_ROOT/requirements.txt
+pip install -r $BUILD_ROOT/test_requirements.txt
+
+# generate the UI
+cd $BUILD_ROOT/ui
+
+case $UN2_BUILD_OS in
```

Comment:
We have in the past treated this differently in Windows than in other platforms.  I did this so that we're prepared if we treated them differently again in the future.  I could simple take it out, since it isn't needed currently.